modin/core/dataframe/base/dataframe/utils.py

```diff
@@ -48,3 +53,71 @@ class JoinType(Enum):  # noqa: PR01
     LEFT = "left"
     RIGHT = "right"
     OUTER = "outer"
+
+
+def join_columns(
+    left: pandas.Index,
+    right: pandas.Index,
+    left_on: Union[Iterable[IndexLabel], IndexLabel],
+    right_on: Union[Iterable[IndexLabel], IndexLabel],
+    suffixes: Tuple[str, str],
+) -> Tuple[pandas.Index, Dict[IndexLabel, IndexLabel], Dict[IndexLabel, IndexLabel]]:
+    """
+    Compute resulting columns for the two dataframes being merged.
+
+    Parameters
+    ----------
+    left : pandas.Index
+        Columns of the left frame to join.
+    right : pandas.Index
+        Columns of the right frame to join.
+    left_on : list-like or scalar
+        Column names on which the frames are joined in the left DataFrame.
+    right_on : list-like or scalar
+        Column names on which the frames are joined in the right DataFrame.
+    suffixes : tuple[str, str]
+        A 2-length sequence containing suffixes to append to the intersected columns.
+
+    Returns
+    -------
+    pandas.Index, dict[IndexLabel -> IndexLabel], dict[IndexLabel -> IndexLabel]
+        Returns columns for the resulting frame and mappings of old to new column
+        names for `left` and `right` accordingly.
+    """
+    left_on = cast(Iterable[IndexLabel], [left_on] if is_scalar(left_on) else left_on)
```

Comment:
Wouldn't it be enough?
```suggestion
    left_on: Iterable[IndexLabel] = [left_on] if is_scalar(left_on) else left_on
```

Reply:
Nope, it doesn't work
```
error: Name "left_on" already defined on line 57  [no-redef]
error: Name "right_on" already defined on line 57  [no-redef]
```

Reply:
That is, it does not allow changing the type of a variable at all?

Reply:
it allows to do explicit casts, but it raises and error on any re-definition

Reply:
> it allows to do explicit casts

it also verifies whether it's possible to make the requested type conversion:
```python
var : Union[Hashable, Sequence[Hashable]] = x
var = cast(Iterable[Hashable], var) # doesn't allow as Iterable and Sequence API doesn't match
var = cast(Sequence[Hashable], var) # works fine
```

Reply:
> > it allows to do explicit casts
> 
> it also verifies whether it's possible to make the requested type conversion:
> 
> ```python
> var : Union[Hashable, Sequence[Hashable]] = x
> var = cast(Iterable[Hashable], var) # doesn't allow as Iterable and Sequence API doesn't match
> var = cast(Sequence[Hashable], var) # works fine
> ```

Interesting.

Will the error persist if just `IndexLabel` is used.

Reply:
`Union[Hashable, Sequence[Hashable]]` is `IndexLabel`, so yes, the error message will be the same

Reply:
thanks for checking that